hetmatpy/pipeline.py

```diff
@@ -8,6 +8,22 @@
 import hetmatpy.degree_weight
 import hetmatpy.hetmat
 
+FLOAT_ERROR_TOLERANCE = 1e-5
+
+
+def calculate_sd(sum_of_squares, unsquared_sum, number_nonzero):
+    """
+    Calculate the standard deviation and validate the incoming data
+    """
+    if number_nonzero == 1:
+        return 0
```

Comment:
My understanding is that this should be None / NaN due to divide by zero https://stackoverflow.com/a/8023909/4651668

Why break it out as a special case?

Reply:
I pulled it out as a special case to prevent division by zero in calculating the standard deviation.

You're right though, it should be returning None instead of 0. I'll update this, and modify the code in add_gamma_hurdle to look for standard deviations of None in addition to 0.